docs/ref/contrib/admin/filters.txt

```diff
@@ -0,0 +1,188 @@
+                    return super().lookups(request, model_admin)
+
+            def queryset(self, request, queryset):
+                if request.user.is_superuser:
+                    return super().queryset(request, queryset)
+
+    Also as a convenience, the ``ModelAdmin`` object is passed to the
+    ``lookups`` method, for example if you want to base the lookups on the
+    available data::
+
+        class AdvancedDecadeBornListFilter(DecadeBornListFilter):
+
+            def lookups(self, request, model_admin):
+                """
+                Only show the lookups if there actually is
+                anyone born in the corresponding decades.
+                """
+                qs = model_admin.get_queryset(request)
+                if qs.filter(birthday__gte=date(1980, 1, 1),
+                            birthday__lte=date(1989, 12, 31)).exists():
+                    yield ('80s', _('in the eighties'))
+                if qs.filter(birthday__gte=date(1990, 1, 1),
+                            birthday__lte=date(1999, 12, 31)).exists():
+                    yield ('90s', _('in the nineties'))
+
+Using a field name and an explicit ``FieldListFilter``
+======================================================
+
+Finally, if you wish to specify an explicit filter type to use with a field you
+may provide a ``list_filter`` item as a tuple, where the first element is a
+field name and the second element is a class inheriting from
+``django.contrib.admin.FieldListFilter``, for example::
+
+    class PersonAdmin(admin.ModelAdmin):
+        list_filter = (
+            ('is_staff', admin.BooleanFieldListFilter),
+        )
+
+Here the ``is_staff`` field will use the ``BooleanFieldListFilter``. Specifying
+only the field name, fields will automatically use the appropriate filter for
+most cases, but this format allows you to control the filter used.
+
+There are some available classes that will not be used automatically.
+
+You can limit the choices of a related model to the objects involved in
+that relation using ``RelatedOnlyFieldListFilter``::
+
+    class BookAdmin(admin.ModelAdmin):
+        list_filter = (
+            ('author', admin.RelatedOnlyFieldListFilter),
+        )
+
+Assuming ``author`` is a ``ForeignKey`` to a ``User`` model, this will
+limit the ``list_filter`` choices to the users who have written a book,
+instead of listing all users.
+
+You can filter empty values using ``EmptyFieldListFilter``, which can
+filter on both empty strings and nulls, depending on what the field
+allows to store::
+
+    class BookAdmin(admin.ModelAdmin):
+        list_filter = (
+            ('title', admin.EmptyFieldListFilter),
+        )
+
+.. note::
+
+    The ``FieldListFilter`` API is considered internal and might be changed.
+
+.. note::
+
+    The :class:`~django.contrib.contenttypes.fields.GenericForeignKey` field is
+    not supported.
+
+List filter's typically appear only if the filter has more than one choice. A
```

Comment:
Again, extant documentation, but shouldn't this be `List filters`, or is it supposed to be `A List Filter's`?